FabLabKasse/gui.py

```diff
@@ -570,14 +584,14 @@ def askUser():
                     # TOOD show amount returned on receipt (needs some rework, because it is not yet stored in the order and so we cannot re-print receipts)
                     self.shoppingBackend.print_receipt(paymentmethod.receipt_order_id)
                 except PrinterError,  e:
-                    QtGui.QMessageBox.warning(self, "Quittung", "Drucker scheint offline zu sein."+
-                        "\nFalls du wirklich eine Quittung brauchst, melde dich bei "+
+                    QtGui.QMessageBox.warning(self, "Quittung", "Drucker scheint offline zu sein." +
+                        "\nFalls du wirklich eine Quittung brauchst, melde dich bei " +
                         "kasse@fablab.fau.de mit Datum, Uhrzeit und Betrag.")
                     logging.warning("printing receipt failed: {}".format(repr(e)))
         if paymentmethod.successful:
             paymentmethod.show_thankyou()
-            self.shoppingBackend.set_current_order(None)
```

Comment:
set_current_order(None): switch away from the current cart to "nothing selected". You could possibly switch back to the old order if it was not deleted.
delete_current_order(): delete the current order. Not possible if an order is already paid, as you don't want to delete it from the database. (Should imply set_current_order(None))